src/chatd.cpp

```diff
@@ -4421,6 +4490,47 @@ void Chat::handleRetentionTime()
     {
         CALL_LISTENER(onUnreadChanged);
     }
+    return nextRetentionHistCheck(updateTimer);
+}
+
+void Chat::getIdxByRetentionTime(Idx &idx)
+{
+    time_t expireRetentionTs = time(nullptr) - mRetentionTime;
+    for (Idx i = highnum(); i >= lownum(); i--)
+    {
+        if (at(i).ts <= expireRetentionTs)
+        {
+            idx = i;
+            return;
+        }
+    }
+
+    if (lownum() == mOldestIdxInDb)
+    {
+        idx = CHATD_IDX_INVALID;
+        return;
+    }
+
+    CALL_DB(getIdxByRetentionTime, expireRetentionTs, idx);
+}
+
+time_t Chat::nextRetentionHistCheck(bool updateTimer)
```

Comment:
Sorry, this method was refactorized, and I forgot to reestablish the original condition.
We don't have to call updateRetentionCheckPeriod if updateTimer is false, to avoid an infitite loop
when handleRetentionTime upon timeour expiration.

Commit uploaded: f50f91c33b0e645ce04b0212e134e092cfc4ce3b